packages/kbn-dev-utils/src/plugin_list/discover_plugins.ts

```diff
@@ -31,21 +31,43 @@ export interface Plugin {
   relativeDir?: string;
   relativeReadmePath?: string;
   readmeSnippet?: string;
+  readmeAsciidocAnchor?: string;
 }
 
 export type Plugins = Plugin[];
 
 const getReadmeName = (directory: string) =>
   Fs.readdirSync(directory).find((name) => name.toLowerCase() === 'readme.md');
 
+const getReadmeAsciidocName = (directory: string) =>
+  Fs.readdirSync(directory).find((name) => name.toLowerCase() === 'readme.asciidoc');
+
 export const discoverPlugins = (pluginsRootDir: string): Plugins =>
   simpleKibanaPlatformPluginDiscovery([pluginsRootDir], []).map(
     ({ directory, manifest: { id } }): Plugin => {
       const readmeName = getReadmeName(directory);
+      const readmeAsciidocName = getReadmeAsciidocName(directory);
 
       let relativeReadmePath: string | undefined;
       let readmeSnippet: string | undefined;
-      if (readmeName) {
+      let readmeAsciidocAnchor: string | undefined;
+
+      if (readmeAsciidocName) {
+        const readmePath = Path.resolve(directory, readmeAsciidocName);
+        relativeReadmePath = Path.relative(REPO_ROOT, readmePath);
+
+        const readmeText = Fs.readFileSync(relativeReadmePath).toString();
+        // First group is to grab the anchor - \[\[(.*)\]\]
+        // Tecond group, (== ), removes the equals from the header
+        // Third group could perhaps be done better, but is essentially:
+        // If there is a sub heading after the intro, match the intro and stop - (([\s\S]*?)(?=\=\=\=|\[\[)))
+        // If there is not a sub heading after the intro, match the intro - ([\s\S]*)
+        const matchAnchorAndIntro = /\[\[(.*)\]\]\n(== )(((([\s\S]*?)(?=\=\=\=|\[\[)))|([\s\S]*))/gm;
```

Comment:
I'm sure you'd get there, just want to request that this be broken into a helper function and unit tested with inline snapshots if you don't mind 😄